Tests/BugsnagAppTest.m

```diff
@@ -0,0 +1,153 @@
+//
+//  BugsnagDeviceTest.m
+//  Tests
+//
+//  Created by Jamie Lynch on 01/04/2020.
+//  Copyright © 2020 Bugsnag. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+
+#import "BugsnagAppWithState.h"
+#import "BugsnagConfiguration.h"
+#import "BugsnagTestConstants.h"
+
+@interface BugsnagApp ()
++ (BugsnagApp *)appWithDictionary:(NSDictionary *)event config:(BugsnagConfiguration *)config;
+
+- (NSDictionary *)toDict;
+@end
+
+@interface BugsnagAppWithState ()
++ (BugsnagAppWithState *)appWithDictionary:(NSDictionary *)event config:(BugsnagConfiguration *)config;
++ (BugsnagAppWithState *)appWithOomData:(NSDictionary *)event;
+- (NSDictionary *)toDict;
+@end
+
+@interface BugsnagAppTest : XCTestCase
+@property NSDictionary *data;
+@property BugsnagConfiguration *config;
+@end
+
+@implementation BugsnagAppTest
+
+- (void)setUp {
+    [super setUp];
+    self.data = @{
+            @"system": @{
+                    @"application_stats": @{
+                            @"active_time_since_launch": @2,
+                            @"background_time_since_launch": @5,
+                            @"application_in_foreground": @YES,
+                    },
+                    @"CFBundleExecutable": @"MyIosApp",
+                    @"CFBundleIdentifier": @"com.example.foo.MyIosApp",
+                    @"CFBundleShortVersionString": @"5.6.3",
+                    @"CFBundleVersion": @"1",
+                    @"app_uuid": @"dsym-uuid-123"
+            },
+            @"user": @{
+                    @"config": @{
+                            @"releaseStage": @"beta"
+                    }
+            }
+    };
```

Comment:
v. brief commentary explaining that this is a mock KSCrash report would be useful.

Reply:
Added a brief comment here